packages/salesforcedx-vscode-soql/src/client/client.ts

```diff
@@ -60,28 +53,7 @@ export async function startLanguageClient(
       configurationSection: 'soql',
       fileEvents: workspace.createFileSystemWatcher('**/*.soql')
     },
-    middleware: {
-      // The SOQL LSP server may include special completion items as "placeholders" for
-      // the client to expand with information from the users' default Salesforce Org.
-      // We do that here as middleware, transforming the server response before passing
-      // it up to VSCode.
-      provideCompletionItem: async (
-        document,
-        position,
-        context,
-        token,
-        next
-      ) => {
-        const items = (await next(
-          document,
-          position,
-          context,
-          token
-        )) as ProtocolCompletionItem[];
-
-        return expandPlaceholders(items);
-      }
-    }
```

Comment:
this shouldn't be part of this PR right?

Reply:
It is... I've separated the completion vs. query validation into different files. `client.ts` was getting too big and with many responsibilities (initialization, completion, query validation)